pkg/chunk/purger/purger.go

```diff
@@ -714,3 +691,56 @@ func makeDeleteRequestWithLogger(deleteRequest DeleteRequest, l log.Logger) dele
 	logger := log.With(l, "user_id", deleteRequest.UserID, "request_id", deleteRequest.RequestID)
 	return deleteRequestWithLogger{deleteRequest, logger}
 }
+
+// inProcessRequestsCollection stores DeleteRequests which are in process by each user.
+// Currently we only allow processing of one delete request per user so it stores single DeleteRequest per user.
+type inProcessRequestsCollection struct {
+	requests map[string]*DeleteRequest
+	mtx      sync.RWMutex
+}
+
+func newInProcessRequestsCollection() *inProcessRequestsCollection {
+	return &inProcessRequestsCollection{requests: map[string]*DeleteRequest{}}
+}
+
+func (i *inProcessRequestsCollection) add(userID string, request *DeleteRequest) {
```

Comment:
I would call it `set()` because overrides any previous request (if any) and doesn't "add" the request to a list.